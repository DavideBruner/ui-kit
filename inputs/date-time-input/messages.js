import { defineMessages } from 'react-intl';

export default defineMessages({
  labelRange: {
    id: 'UIKit.DateTimeInput.labelRange',
    description: 'Label for range inputs',
    defaultMessage: 'to',
  },
});
