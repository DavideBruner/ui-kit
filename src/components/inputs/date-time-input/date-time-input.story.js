import React from 'react';
import { storiesOf } from '@storybook/react';
import { IntlProvider } from 'react-intl';
import { action } from '@storybook/addon-actions';
import withReadme from 'storybook-readme/with-readme';
import { withKnobs, boolean, text, select } from '@storybook/addon-knobs';
import { Value } from 'react-value';
import Section from '../../../../.storybook/decorators/section';
import Readme from './README.md';
import DateTimeInput from './date-time-input';

storiesOf('Inputs', module)
  .addDecorator(withKnobs)
  .addDecorator(withReadme(Readme))
  .add('DateTimeInput', () => {
    const locale = select('locale', ['de', 'en'], 'en');
    const timeZone = select(
      'timeZone',
      ['Europe/Madrid', 'America/Los_Angeles'],
      'Europe/Madrid'
    );
    return (
      <Section>
        <IntlProvider locale={locale}>
          <Value
            key={`${locale}-${timeZone}`}
            defaultValue="2017-12-31T16:02:50.000Z"
            render={(value, onChange) => (
              <DateTimeInput
                key={`${locale}-${timeZone}`}
                id={text('id', '')}
                placeholder={text('placeholder', 'Select a date...')}
                mode={select('mode', ['single', 'multiple', 'range'], 'single')}
                isDisabled={boolean('isDisabled', false)}
                value={text('value (Date in UTC)', value)}
                onChange={datetime => {
                  action('onChange')(datetime);
                  onChange(datetime);
                }}
                isInvalid={boolean('isInvalid?', false)}
                timeZone={timeZone}
                horizontalConstraint={select(
                  'horizontalConstraint',
                  ['xs', 's', 'm', 'l', 'xl', 'scale'],
                  'm'
                )}
              />
            )}
          />
        </IntlProvider>
      </Section>
    );
  });
