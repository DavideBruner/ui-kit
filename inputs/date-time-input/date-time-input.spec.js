import React from 'react';
import { shallow } from 'enzyme';
import isTouchDevice from 'is-touch-device';
import { DateTimeInput } from './date-time-input';

jest.mock('flatpickr');
jest.mock('is-touch-device');

// user is in London
// eslint-disable-next-line no-extend-native
Date.prototype.getTimezoneOffset = () => 0;

// but she wants to see time in Mardid time zone
const timeZone = 'Europe/Madrid';

const createTestProps = custom => ({
  onChange: jest.fn(),
  value: null,
  placeholder: 'test',
  intl: { formatMessage: jest.fn(message => message.id), locale: 'en' },
  timeZone,
  ...custom,
});

describe('<DateTimeInput />', () => {
  beforeEach(() => {
    isTouchDevice.mockClear();
    isTouchDevice.mockReturnValue(false);
  });
  it('should render', () => {
    expect(shallow(<DateTimeInput {...createTestProps()} />)).toMatchSnapshot();
  });
});
