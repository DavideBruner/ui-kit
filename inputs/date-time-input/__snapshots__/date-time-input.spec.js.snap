// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateTimeInput /> should render 1`] = `
<Horizontal
  constraint="scale"
>
  <div
    className="container"
    onMouseOver={[Function]}
  >
    <DatePickerBody
      formattedValue={null}
      horizontalConstraint="scale"
      isDisabled={false}
      isInvalid={false}
      numberOfFormattedValueChars={16}
      onClearPicker={[Function]}
      placeholder="test"
      timeScale="datetime"
    />
  </div>
</Horizontal>
`;
